docker/apm-server/managed/main.go

```diff
@@ -155,31 +163,33 @@ func apmPackagePolicy(policyID string, pkg eprPackage) packagePolicy {
 
 type kibanaClient struct {
 	fleetURL string
-	pkgURL   string
 }
 
 func newKibanaClient() *kibanaClient {
 	host := os.Getenv("KIBANA_HOST")
 	if host == "" {
-		host = "http://admin:changeme@localhost:5601"
-	}
-	pkgURL := os.Getenv("XPACK_FLEET_REGISTRYURL")
-	if pkgURL == "" {
-		pkgURL = "https://epr.elastic.co"
-	}
-	return &kibanaClient{
-		fleetURL: fmt.Sprintf("%s/api/fleet", host),
-		pkgURL:   pkgURL,
+		host = "http://admin:changeme@kibana:5601"
 	}
+	return &kibanaClient{fleetURL: fmt.Sprintf("%s/api/fleet", host)}
 }
 
-func (client *kibanaClient) getPackages(query string) ([]eprPackage, error) {
-	url := fmt.Sprintf("%s/search?%s", client.pkgURL, query)
-	var packages []eprPackage
-	if err := makeRequest(http.MethodGet, url, nil, &packages); err != nil {
-		return packages, errors.Wrap(err, "getPackages")
+func (client *kibanaClient) getAPMPackage() (*eprPackage, error) {
+	url := fmt.Sprintf("%s/epm/packages?experimental=true", client.fleetURL)
+	var pkgs eprPackagesResponse
+	if err := makeRequest(http.MethodGet, url, nil, &pkgs); err != nil {
+		return nil, err
+	}
+	for _, p := range pkgs.Packages {
+		if p.Name != "apm" {
+			continue
+		}
+		var apm eprPackageResponse
+		url := fmt.Sprintf("%s/epm/packages/%s-%s", client.fleetURL, p.Name, p.Version)
+		err := makeRequest(http.MethodGet, url, nil, &apm)
+		return &apm.Package, err
+
 	}
-	return packages, nil
+	return nil, nil
```

Comment:
nit:
 
```suggestion
	return nil, errors.New("no apm package found")
```

then you can get rid of the nil check in the caller